backend_with_db.js

```diff
@@ -2,11 +2,15 @@ import express from "express";
 import cors from "cors";
 import bcrypt from "bcrypt";
 import jwt from "jsonwebtoken";
-
+import dotenv from "dotenv";
+import mongoose from "mongoose";
+import { body, validationResult } from "express-validator";
+import buildingRoutes from './routes/buildings.js';
 import userServices from "./models/user-services.js";
 
+dotenv.config();
 const app = express();
-const port = 8000;
+const port = process.env.PORT || 8000;
```

Comment:
The code breaks without the ENV, so there's no real reason to try and give it options to not have it - though I understand your intention this is probably unecessary.